weechat-relay/src/main/java/com/ubergeek42/weechat/relay/RelayConnection.java

```diff
@@ -187,6 +193,8 @@ public void run() {
                     }
 
                     if (data_pos == data.length) {
+                        messageReceivedOnCurrentConnection = true;
```

Comment:
Perhaps a better place to handle this would be in `onAuthenticated`. This is called when the result of `info` is returned. My testing showed that WeeChat doesn't close the connection on `init`, but rather any subsequent command. In that case, the tri-state could represent whether we have `authenticated`.